react-components/src/components/RevealToolbar/SelectSingle3DModelButton.tsx

```diff
@@ -0,0 +1,85 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+import { useState, type ReactElement } from 'react';
+
+import { Button, Dropdown, Menu, Tooltip as CogsTooltip } from '@cognite/cogs.js';
+import { useTranslation } from '../i18n/I18n';
+import styled from 'styled-components';
+import { useAll3dModels } from '../../hooks/useAll3dModels';
+import { useSDK } from '../RevealCanvas/SDKProvider';
+import { RevealContext } from '../RevealContext/RevealContext';
+import { type CogniteClient } from '@cognite/sdk';
+import { ModelsList } from './ModelsList';
+import { useRevisions } from '../../hooks/useRevisions';
+import { type ModelWithRevision } from '../../hooks/types';
+
+type SelectSingle3DModelButtonProps = {
+  onSingleModelChanged?: (model: ModelWithRevision | undefined) => void;
+};
+
+export const SelectSingle3DModelButton = ({
+  onSingleModelChanged
+}: SelectSingle3DModelButtonProps): ReactElement => {
+  const sdk = useSDK();
+
+  const handleSingleModelChange = (model: ModelWithRevision | undefined): void => {
+    if (onSingleModelChanged !== undefined) onSingleModelChanged(model);
+  };
+
+  return (
+    <RevealContext sdk={sdk}>
+      <Single3DModelSelection sdk={sdk} onModelChange={handleSingleModelChange} />
+    </RevealContext>
+  );
+};
+
+type Single3DModelSelectionProps = {
+  sdk: CogniteClient;
+  onModelChange: (model: ModelWithRevision | undefined) => void;
+};
+
+export const Single3DModelSelection = ({
```

Comment:
Should be in separate file

Reply:
https://github.com/cognitedata/reveal/pull/4746/commits/f51740e72efab7b49df11e4dc584ae056644e8f9